tools/server-spring/src/main/java/org/eclipse/rdf4j/http/server/repository/namespaces/NamespaceController.java

```diff
@@ -38,11 +43,13 @@
  * @author Arjohn Kampman
  */
 public class NamespaceController extends AbstractController {
+	private static final String URL_REGEX = "^(((https?|ftps?|gopher|telnet|nntp)://)(%[0-9A-Fa-f]{2}|[-()_.!~*';*/?:@&#=+$,A-Za-z0-9])+)([).!';/?:,][[:blank:]])?$";
+	private static final Pattern URL_PATTERN = Pattern.compile(URL_REGEX);
```

Comment:
This is too strict - it does not allow for URIs other than the ones starting with the schemes you have in there, for one thing. 

It's probably better (for consistency's sake) to leverage the RDF4J `ParsedIRI` class here, which can be used to check that a string is a valid IRI.

Reply:
I found this code snipped which validates the base URI. I think it would make sense to have the same validation for the namespace URI too. 
https://github.com/eclipse/rdf4j/blob/b849630bfac9c84e0444390179409456da043b15/core/repository/sparql/src/main/java/org/eclipse/rdf4j/repository/sparql/query/SPARQLOperation.java#L54

..or I could just check if the namespace URI  isn't empty.